### PR TITLE
[@type/elasticsearch] Support import HttpConnector with ES6 syntax (for Angular 5 importing)

### PR DESCRIPTION
**Pull request by panoti**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/elasticsearch/src/lib/connectors/http.d.ts [hunk 1]
```diff
@@ -11,3 +11,6 @@ declare class HttpConnector {
     makeReqParams(params: any): any;
     request(params: any, callback: (error: any, response: any, status: any, headers: any) => void): void;
 }
+
+declare namespace HttpConnector {
```
Comment (rbuckton) [inline types/elasticsearch/src/lib/connectors/http.d.ts]: By merging a namespace this actually lies to the type system that `HttpConnector` can be imported using `import * as`. Is there an issue with importing using `import =` with Angular 5?

Comment (panoti) [reply types/elasticsearch/src/lib/connectors/http.d.ts]: I use lastest version of Angular is 5, may be other version from version 2 catch the same error. They use es6 syntax `import ... from ...` while node js use commonjs with `const ... = require(...)`. I read at https://stackoverflow.com/questions/39415661/what-does-resolves-to-a-non-module-entity-and-cannot-be-imported-using-this. I just add one line for angular (I found momentjs also declare too) and node js can work together.

Comment (panoti) [reply types/elasticsearch/src/lib/connectors/http.d.ts]: I tried with `import =` but it work incorrectly. Look at here `module.exports = HttpConnector;` in `elasticsearchjs` lib, I think this is good solution now, don't you?

Comment (panoti) [reply types/elasticsearch/src/lib/connectors/http.d.ts]: sorry, wait me re-code, some update from angular 6 and elasticsearch 15 make me confuse.

**Comment by rbuckton** [comment]
Per https://github.com/DefinitelyTyped/DefinitelyTyped#should-i-add-an-empty-namespace-to-a-package-that-doesnt-export-a-module-to-use-es6-style-imports, this change is a practice that we discourage.

**Comment by panoti** [comment]
@rbuckton OK, I will fix it.

**Review by armanio123 — CHANGES_REQUESTED** [review]
As per @rbuckton comments I'm going to mark this PR as "request changes" for tracking purposes.

We'll review again when the comments have been addressed.

**Comment by minestarks** [comment]
@panoti Do you plan to address the requested changes? Please do so before we close the PR for inactivity. Thanks!

**Pull request closed without merging**